File: app/src/main/java/br/edu/farol/gadoplus/adapter/PesagemAnimalAdapter.java
package br.edu.farol.gadoplus.adapter;

import android.app.Application;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;
import br.edu.farol.gadoplus.R;
import br.edu.farol.gadoplus.model.Animal;
import br.edu.farol.gadoplus.model.PesagemAnimal;
import br.edu.farol.gadoplus.storage.database.repository.AnimalRepository;

public class PesagemAnimalAdapter extends RecyclerView.Adapter<PesagemAnimalAdapter.PesagemAnimalViewHolder> {

    private List<PesagemAnimal> pesagemAnimals;
    private OnItemClickListener listener;
    private AnimalRepository animalRepository;
    private Application application;

    @Override
    public PesagemAnimalViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.recyclerview_item, parent, false);
        animalRepository =  new AnimalRepository(application);
        return new PesagemAnimalViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(PesagemAnimalViewHolder viewHolder, int i) {

        viewHolder.nome.setText(pesagemAnimals.get(i).getPeso()+" Kg");

        try{
            Animal animal = animalRepository.getById(pesagemAnimals.get(i).getAnimalId());
            viewHolder.descricao.setText("Animal: " + animal.getNome());
        }catch (Exception e){
            System.out.println(e);
        }
    }

    @Override
    public int getItemCount() {
        if (pesagemAnimals != null)
            return pesagemAnimals.size();
        else return 0;
    }

    public void setPesagemAnimal(List<PesagemAnimal> pesagemAnimals) {
        this.pesagemAnimals = pesagemAnimals;
        notifyDataSetChanged();
    }

    public void setApplication(Application application){
        this.application = application;
    }

    class PesagemAnimalViewHolder extends RecyclerView.ViewHolder {
        private TextView nome, descricao, hectares;

        public PesagemAnimalViewHolder(View itemView) {
            super(itemView);
            nome = itemView.findViewById(R.id.text_view_title);
            descricao = itemView.findViewById(R.id.text_view_description);
            hectares = itemView.findViewById(R.id.text_view_priority);

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    int position = getAdapterPosition();
                    if (listener != null && position != RecyclerView.NO_POSITION)
                        listener.onItemClick(pesagemAnimals.get(position));
                }
            });
        }
    }

    public interface OnItemClickListener {
        void onItemClick(PesagemAnimal pesagemAnimal);
    }

    public void setOnClickListener(OnItemClickListener listener) {
        this.listener = listener;
    }


}

File: app/src/main/java/br/edu/farol/gadoplus/adapter/PesagemAnimalAdapter.kt
package br.edu.farol.gadoplus.adapter

import android.app.Application
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import br.edu.farol.gadoplus.R
import br.edu.farol.gadoplus.adapter.PesagemAnimalAdapter.PesagemAnimalViewHolder
import br.edu.farol.gadoplus.model.PesagemAnimal
import br.edu.farol.gadoplus.storage.database.repository.AnimalRepository

class PesagemAnimalAdapter : RecyclerView.Adapter<PesagemAnimalViewHolder>() {
    private var pesagemAnimals: List<PesagemAnimal>? = null
    private var listener: OnItemClickListener? = null
    private var animalRepository: AnimalRepository? = null
    private var application: Application? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PesagemAnimalViewHolder {
        val itemView = LayoutInflater.from(parent.context)
                .inflate(R.layout.recyclerview_item, parent, false)
        animalRepository = AnimalRepository(application)
        return PesagemAnimalViewHolder(itemView)
    }

    override fun onBindViewHolder(viewHolder: PesagemAnimalViewHolder, i: Int) {
        viewHolder.nome.setText("${pesagemAnimals!![i].peso}   Kg")
        try {
            val animal = animalRepository!!.getById(pesagemAnimals!![i].animalId)
            viewHolder.descricao.text = "Animal: " + animal.nome
        } catch (e: Exception) {
            println(e)
        }
    }

    override fun getItemCount(): Int {
        return if (pesagemAnimals != null) pesagemAnimals!!.size else 0
    }

    fun setPesagemAnimal(pesagemAnimals: List<PesagemAnimal>?) {
        this.pesagemAnimals = pesagemAnimals
        notifyDataSetChanged()
    }

    fun setApplication(application: Application?) {
        this.application = application
    }

    inner class PesagemAnimalViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val nome: TextView
        val descricao: TextView
        private val hectares: TextView

        init {
            nome = itemView.findViewById(R.id.text_view_title)
            descricao = itemView.findViewById(R.id.text_view_description)
            hectares = itemView.findViewById(R.id.text_view_priority)
            itemView.setOnClickListener {
                val position = adapterPosition
                if (listener != null && position != RecyclerView.NO_POSITION) listener!!.onItemClick(pesagemAnimals!![position])
            }
        }
    }

    interface OnItemClickListener {
        fun onItemClick(pesagemAnimal: PesagemAnimal?)
    }

    fun setOnClickListener(listener: OnItemClickListener?) {
        this.listener = listener
    }
}